deltachat-ffi/src/lib.rs

```diff
@@ -1192,17 +1191,20 @@ pub unsafe extern "C" fn dc_archive_chat(
         return;
     }
     let ffi_context = &*context;
-    let archive = if archive == 0 {
-        false
-    } else if archive == 1 {
-        true
-    } else {
-        return;
+    let archive_state = match archive {
+        0 => ArchiveState::Normal,
+        1 => ArchiveState::Archived,
+        2 => ArchiveState::Pinned,
```

Comment:
This should use something like `ArchiveState::try_from(archive)` instead

Reply:
https://doc.rust-lang.org/std/convert/trait.TryFrom.html

Reply:
maybe it is okay, as long as this is the only place where this converting takes place.